ddtrace/auto.py

```diff
@@ -1 +1,21 @@
+"""
+.. _ddtraceauto:
+
+Importing ``ddtrace.auto`` installs Datadog instrumentation in the runtime. It should be used
+when :ref:`ddtrace-run<ddtracerun>` is not an option. Using it with :ref:`ddtrace-run<ddtracerun>`
+is unsupported and may lead to undefined behavior::
```

Comment:
🤔 interesting..... I don't think I've given this much thought before, and I think I would have assumed this "just worked"... but....

`ddtrace-run` loads `ddtrace.bootstrap.sitecustomize` as the `sitecustomize` module, but then `ddtrace.auto` loads it as `ddtrace.bootstrap.sitecustomize` so it gets reloaded since the `sys.module` cache has `sitecustomize` not `ddtrace.bootstrap.sitecustomize`.

I mean, this makes complete sense, but I hadn't stopped to think about it before.

```python
❯ ddtrace-run python
Python 3.11.0 (main, Feb 22 2023, 15:34:50) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> sys.modules["sitecustomize"]
<module 'sitecustomize' from '/Users/brett.langdon/datadog/dd-trace-py/ddtrace/bootstrap/sitecustomize.py'>
>>> import ddtrace.auto
>>> sys.modules["ddtrace.bootstrap.sitecustomize"]
<module 'ddtrace.bootstrap.sitecustomize' from '/Users/brett.langdon/datadog/dd-trace-py/ddtrace/bootstrap/sitecustomize.py'>
>>> sys.modules["sitecustomize"] is sys.modules["ddtrace.bootstrap.sitecustomize"]
False
>>>
```

Reply:
"undefined behavior" here really just means that I haven't given it much thought and it's not how the interface is intended to be used.